plugins/modules/user_info.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: user_info
+
+author:
+  - Polona Mihalič (@PolonaM)
+short_description: Returns information about the users.
+description:
+  - Returns information about the users.
+version_added: 1.2.0
+extends_documentation_fragment:
+  - scale_computing.hypercore.cluster_instance
+seealso:
+  - module: scale_computing.hypercore.user
+"""
+
+EXAMPLES = r"""
+- name: List all users
+  scale_computing.hypercore.user_info:
+  register: users
+"""
+
+RETURN = r"""
+records:
+  description:
+    - A list of user records.
+  returned: success
+  type: list
+  sample:
+    - fullname: xlab
+      role_uuids:
+        - 38b346c6-a626-444b-b6ab-92ecd671afc0
+        - 7224a2bd-5a08-4b99-a0de-9977089c66a4
```

Comment:
Role uuid does not tell much. I guess later we will need to add also a role name. But dedicated role_info module is out of scope in this phase, so I want only to have format that will allow future modification in backward compatible way.

If today we output:
```
  sample:
    - fullname: xlab
      roles:
        - uuid: 38b346c6-a626-444b-b6ab-92ecd671afc0
        - uuid: 7224a2bd-5a08-4b99-a0de-9977089c66a4
```

And in future we can extend with (role) name:
```
  sample:
    - fullname: xlab
      roles:
        - uuid: 38b346c6-a626-444b-b6ab-92ecd671afc0
          name: admin
        - uuid: 7224a2bd-5a08-4b99-a0de-9977089c66a4
          name: backup
```

Reply:
Update @PolonaM: please include role name in user_info output. Also when creating a new user, roles will need to be selected, and role will be identified by name.